File: C++/bee 1045.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

using namespace std;

void bubbleSort(float arr[], int len) {
    len--;
    for (int i = 0; i < len; i++) {
        for (int j = 0; j < len; j++) {
            if (arr[j] < arr[j+1]) {
                int aux = arr[j];
                arr[j] = arr[j+1];
                arr[j+1] = aux;
            }
        }
    }
}

int main() {

    float a, b, c;
    cin >> a >> b >> c;
    float lados[] = {a,b,c};
    bubbleSort(lados,3);

    if (lados[0] >= lados[1] + lados[2]) {
        cout << "NAO FORMA TRIANGULO" << endl;
        return 0;
    } else if (pow(lados[0],2) == pow(lados[1],2) + pow(lados[2],2)) {
        cout << "TRIANGULO RETANGULO" << endl;
    } else if (pow(lados[0],2) > pow(lados[1],2) + pow(lados[2],2)) {
        cout << "TRIANGULO OBTUSANGULO" << endl;
    } else if (pow(lados[0],2) < pow(lados[1],2) + pow(lados[2],2)) {
        cout << "TRIANGULO ACUTANGULO" << endl;
    }

    int contador = 1;
    if (a == b) {
        contador++;
    }
    if (a == c) {
        contador++;
    }
    if (b == c) {
        contador++;
    }
    if (contador == 4){
        cout << "TRIANGULO EQUILATERO" << endl;
    } else if (contador == 2){
        cout << "TRIANGULO ISOSCELES" << endl;
    }

        return 0;
}

File: Lua/bee 1045.lua
function bubbleSort(arr)
    for i = 1, #arr-1, 1 do
        for j = 1, #arr-1, 1 do
            if arr[j] < arr[j+1] then
                local aux = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = aux
            end
        end
    end
end

local arr = {tonumber(io.read("*n")),tonumber(io.read("*n")),tonumber(io.read("*n"))}
bubbleSort(arr)
local a = arr[1]
local b = arr[2]
local c = arr[3]

if a >= b + c then
    print("NAO FORMA TRIANGULO")
elseif a^2 == b^2 + c^2 then
    print("TRIANGULO RETANGULO")
elseif a^2 > b^2 + c^2 then
    print("TRIANGULO OBTUSANGULO")
elseif a^2 < b^2 + c^2 then
    print("TRIANGULO ACUTANGULO")
end

local contador = 0

if a == b then
    contador = contador + 1
end

if a == c then
    contador = contador + 1
end

if b == c then
    contador = contador + 1
end

if contador == 3 then
    print("TRIANGULO EQUILATERO")
elseif contador == 1 then
    print("TRIANGULO ISOSCELES")
end
